54 Nested properties validation

mwpolcik opened the pull request:
This closes #54.

afronski commented on logger/logger.go:
```diff
@@ -117,10 +122,37 @@ func (logger *Logger) log(verbosity Verbosity, message string) {
+	}
+}
+
+// HasValidationErrors : Check if logger contains error messages
```
I believe we do not need function name and a colon here for documentation purposes.

afronski commented on .gitignore:
```diff
@@ -20,3 +20,6 @@ perun
 
 # Local perun configuration:
 .perun
+preprocessed.yml
```
Out of curiosity: what is it?

mwpolcik replied:
This is a temporary file which is being created when `FixFunctions` is called from `intrinsicsolver` package.

afronski commented on .gitignore:
```diff
@@ -20,3 +20,6 @@ perun
 .perun
+preprocessed.yml
+
+.vscode
```
Let's mark it with a comment (what are those files and from which tool).

mwpolcik commented:
Few comments from me:
1. It is a little bit of spaghetti code, so please be careful during review :)
2. Type Map property has limited validation, because most of the item types of maps are PrimitiveTypes(String, Bool). There is only one exception to this which was not covered.
3. I am not sure if this is covered in the commend in fixMultiLineMap function in fixMultiLineMap.go, but this:
```
		split := strings.Split(*line, name)
		if strings.Contains(pLines[idx+1], "-") && split[1] == "" {
```
In most cases  throw an error: **Index out of range**, e.g.
```
      SecurityGroups:
        - !FindInMap [ EnvSettings, !Ref "ENV", SecurityGroupId ]
```

afronski commented:
@jlampar have a look on what @mwpolcik posted here, I think that bit is related to the `instrinsicsolver.go`. Also, I'd love to avoid creating a temporary file if possible (`preprocessed.yml`).

jlampar commented:
I investigated this and enclosing everything in `if idx+1 < len(pLines)` condition does the trick. I'll be updating `instrinsicsolver.go` soon because I am using it for conversion improvements (changing short names to long). This change will be included.
Creating `preprocessed.yml` was proposed for two reasons:
1. To have preprocessed file to investigate any failures in further parsing (if preprocessing is going well, maybe the original template is noted in a strange manner?).
2. To have any solution at the moment. I haven't figured it yet how writing lines to []byte variable would be performed.

afronski commented:
@jlampar, In that case, please prefix `preprocessed.yml` with `.` in order to hide that file on _UNIX_ / _Linux_ based systems.
